dbms/src/Parsers/ASTShowTablesQuery.h

```diff
@@ -3,24 +3,22 @@
 #include <iomanip>
 #include <Parsers/IAST.h>
 #include <Parsers/ASTQueryWithOutput.h>
+#include <Parsers/ASTNamedChildrenHelper.h>
 
 
 namespace DB
 {
 
-
 /** Query SHOW TABLES or SHOW DATABASES
   */
-class ASTShowTablesQuery : public ASTQueryWithOutput
+class DECLARE_SELF_AND_CHILDREN(ASTShowTablesQuery, FROM, LIKE, LIMIT_LENGTH)
```

Comment:
Because many `identifier`, `literal` is extracted from the parser. this makes the implementation of query param unclear(we cannot know which query clause does not support query param because the identifier(May be many other types of AST) already extracted in the parser will not be added to the children of the `IAST`)

The main task of this pr is to normalize `IAST children`. ensure that all syntax nodes are included in children. The current idea is to cut off all `AST` attributed declarations and replace them with their pos in children(Same `ASTSelectQuery` but I'm not sure that makes sense and the way I've refactored this kind of code with macros so far). 

@tavplubix I think this may need some discussion or can you give me some advice?

Reply:
this is macro's declarations : https://github.com/ClickHouse/ClickHouse/pull/7774/files#diff-6334007ce77de116a356e3ffa6b2e6eeR12